corehq/form_processor/submission_post.py

```diff
@@ -71,53 +73,30 @@ def __init__(self, instance=None, attachments=None, auth_context=None,
         assert domain, "'domain' is required"
         assert instance, instance
         assert not isinstance(instance, HttpRequest), instance
+        openrosa_headers = openrosa_headers or {}
         self.domain = domain
-        self.app_id = app_id
-        self.build_id = build_id
         # get_location has good default
         self.location = location or couchforms.get_location()
-        self.received_on = received_on
-        self.date_header = date_header
-        self.submit_ip = submit_ip
-        self.last_sync_token = last_sync_token
-        self.openrosa_headers = openrosa_headers or {}
         self.instance = instance
```

Comment:
It seems dangerous to have this (original and unsanitized) instance set here, and then another instance, which may be replaced with a new/sanitized copy in the pre-process step, stored on the new `form_processor` object.

Maybe this could be done more safely with a much smaller change. Something like
```suggestion
        self.instance = pre_process_form(instance)
```
And outside of this class
```py
from corehq.form_processor.extension_points import get_form_pre_processor

def pre_process_form(instance):
    pre_process = get_form_pre_processor()
    if pre_process is not None:
        instance = pre_process(instance)
    return instance
```

Reply:
offlined and made a new PR https://github.com/dimagi/commcare-hq/pull/28437.
We would still need this to be done at the point when form is really processed. SubmissionPost still rejects the form in some cases https://github.com/dimagi/commcare-hq/pull/28437/files#diff-6c23c5f0d2be8c7538bb6b9c85188c5fR217